Single cell trajectories in Jupyter tutorial (#2470)
galaxyproject/training-material: topics/transcriptomics/tutorials/scrna-JUPYTER-trajectories/tutorial.md

---
layout: tutorial_hands_on

title: Trajectory Analysis using Python (Jupyter Notebook) in Galaxy
subtopic: single-cell
priority: 11
zenodo_link: 'https://zenodo.org/record/4655947'
questions:
- How can I infer lineage relationships between single cells based on their RNA, without a time series?
objectives:
- Execute multiple plotting methods designed to maintain lineage relationships between cells
- Interpret these plots
time_estimation: 2H
key_points:
- Trajectory analysis is less robust than pure plotting methods, as such 'inferred relationships' are a bigger mathematical leap
- As always with single-cell analysis, you must know enough biology to deduce if your analysis is reasonable, before exploring or deducing novel insight
requirements:

> -
>     type: "internal"
>     topic_name: transcriptomics
>     tutorials:
>         - droplet-quantification-preprocessing
>         - scrna-seq-basic-pipeline
>         - galaxy-intro-jupyter

Reviewer: ```suggestion
-
    type: "internal"
    topic_name: transcriptomics
    tutorials:
        - droplet-quantification-preprocessing
        - scrna-seq-basic-pipeline
-
    type: "internal"
    topic_name: galaxy-interface
    tutorials:
        - galaxy-intro-jupyter
```

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Create a new history for this tutorial
> 2. Import the AnnData object from [Zenodo]({{ page.zenodo_link }})
>
>    ```
>    {{ page.zenodo_link }}/files/Trajectories_Instructions.ipynb
>    {{ page.zenodo_link }}/files/Final_cell_annotated_object.h5ad
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
> 3. **Rename** {% icon galaxy-pencil %} the .h5ad object as `Final cell annotated object`
>
> 4. Check that the datatype is `h5ad`
>
>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="h5ad" %}
>

> > 5. **Rename** {% icon galaxy-pencil %} the .ipynb object as `Trajectories_Instructions.ipynb`
> >

Reviewer: ```suggestion
> 5. **Rename** {% icon galaxy-pencil %} the .ipynb object as `Trajectories_Instructions.ipynb`
>
>    {% snippet faqs/galaxy/datasets_rename.md name="Trajectories_Instructions.ipynb" %}
>
```

>
> 2. {% tool [Manipulate AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy0) %} with the following parameters:
>    - {% icon param-file %} *"Annotated data matrix"*: (output of **Manipulate AnnData** {% icon tool %})
>    - *"Function to manipulate the object"*: `Filter observations or variables`
>    - *"What to filter?"*: `Observations (obs)`
>    - *"Type of filtering?"*: `By key (column) values`
>    - *"Key to filter"*: `cell_type`
>    - *"Type of value to filter"*: `Text`
>    - *"Filter"*: `not equal to`
>    - *"Value"*: `Macrophages`
>
> 3. **Rename** {% icon galaxy-pencil %} output h5ad `T-cell_object.h5ad`
>
{: .hands_on}

You should now have `7795` cells, as opposed to the `7874` you started with. You've only removed a few cells (the contaminants!), but it makes a big difference in the next steps.

Take note of what # this dataset is in your history, as you will need that shortly!

## Launching Jupyter

> ### {% icon hands_on %} Hands-on: Downloading the directions
>
> 1. Click the **Trajectories_Instructions.ipynb** dataset and the download {% icon galaxy-save %} button to get the file onto your computer
>
{: .hands_on}

> JupyterLab is a bit like RStudio but for other coding languages. What, you've never heard of RStudio? Then don't worry, just follow the instructions!

Reviewer: ```suggestion
JupyterLab is a bit like RStudio but for other coding languages. What, you've never heard of [RStudio](https://www.rstudio.com/products/rstudio/features/)? Then don't worry, just follow the instructions!
```

> {% icon warning %} Please note: this is only currently available on the [usegalaxy.eu](https://usegalaxy.eu) site.

Reviewer: ```suggestion
{% icon warning %} Please note: this is only currently available on the [usegalaxy.eu](https://usegalaxy.eu) and [usegalaxy.org](https://usegalaxy.org) sites.
```

> ### {% icon hands_on %} Hands-on: Launching JupyterLab
>
> 1. {% tool [Interactive Jupyter Notebook](tool_id=interactive_tool_jupyter_notebook) %} with the following parameters:
>    - *"Do you already have a notebook?"*: `Start with a fresh notebook`
>    - {% icon param-file %} *"Include data into the environment"*:  `Nothing selected`
>

> >  This may take a moment, but once the `Executed notebook` in your dataset is orange, you are up and running!

Reviewer: ```suggestion
>    This may take a moment, but once the `Executed notebook` in your dataset is orange, you are up and running!
```

I think this will reset numbering with the current indentation level.

>
> 4. Either click on the blue `User menu`, or go to the top of the screen and choose `User` and then `Active InteractiveTools`
>
> 5. Click on the newest `Jupyter Interactive Tool`.
>
{: .hands_on}

Welcome!

> ### {% icon warning %} Danger: You can lose data!
> Do NOT delete or close this notebook dataset in your history. YOU WILL LOSE IT!
{: .warning}

> ### {% icon hands_on %} Hands-on: Creating a notebook
>
> 1. Click the **Python 3** icon under **Notebook**

> > ![Python 3 icon](../../images/wab-python3logo.png "Python 3 Button")

Reviewer: ```suggestion
>
>   ![Python 3 icon](../../images/wab-python3logo.png "Python 3 Button")
```
